[dnf5] Implement missing repoquery functions

inknos opened the pull request:
depends on https://github.com/rpm-software-management/libdnf/pull/1202

jrohel commented on dnfdaemon-client/wrappers/dbus_package_wrapper.hpp:
```diff
@@ -0,0 +1,91 @@
+
+    template <typename... Args>
+    typename std::map<Args...>::mapped_type get_data(std::map<Args...> const& map, typename std::map<Args...>::key_type const& key) const {
+        if(!has_key(map, key)) {
```
`!map.contains(key)` can be used instead of `!has_key(map, key)`.
Ok `contains` requires >=C++20. But `map.count(key) == 0` can be used. So we don't need the own `has_key` private method.

But even that is not the good algorithm. Container is searched twice. First, if there is a `key` in it. And then again to get the value of the `key` (`std::at` method).
We can do it in one search.
Something like this: 
```
auto it = map.find(key);
return it != map.end() ? it->second : typename std::map<Args...>::mapped_type();
```

jrohel replied:
And why does the `get_data` method have a` map` input parameter and no direct access to `raw_data`? The method is now independent of any specific object of the `DbusPackageKeyValue` class.
If this is intended, please mark the method as `static`. It's a private method, so your intent may change in the future.

inknos committed:
dnfdaemon: replace tuples with dictionaries in the resolve method

- PackageDbus is now DbusPackageWrapper
- The dbus interface now returns a(ua{sv}) type
- DbusTransactionItem is a struct containing action and a dictionary
- Code formatted using clang

inknos committed:
dnfdemon-cli: implement repoquery missing functions

m-blaha commented on dnfdaemon-client/wrappers/dbus_package_wrapper.cpp:
```diff
@@ -19,14 +19,4 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
-}
-
-}  // namespace dnfdaemon::client
+namespace dnfdaemon::client {}  // namespace dnfdaemon::client
```
Please remove the file completely - it does not have any meaningful content anyway.

inknos replied:
will do

m-blaha commented:
Not asking the change for this PR - but there is room for improvement in the commits structure. A lot of commits looks like "development noise" and could be squashed into one or two of them. The gist of your change will then be easier to track in git history if needed it, and of course to review.
I'm thinking, for example, of a situation where you fix an old `DbusPackageWrapper` constructor that uses old tuple-like structures in one commit, and then remove that constructor entirely in another commit.
Otherwise the change looks good to me and ready to be merged - thanks :)

inknos commented:
Thank you for your comments. The PR is now two commits. Hope it's better to read and review this way 😃
